Templating/FilterTrait.php

```diff
@@ -38,7 +38,7 @@ public function __construct(CacheManager $cache)
      */
     public function filter($path, $filter, array $config = [], $resolver = null, $referenceType = UrlGeneratorInterface::ABSOLUTE_URL)
     {
-        return $this->cache->getBrowserPath(parse_url($path, PHP_URL_PATH), $filter, $config, $resolver, $referenceType);
+        return $this->cache->getBrowserPath($path, $filter, $config, $resolver, $referenceType);
```

Comment:
isn't this also used with local files? if we talk with a filesystem, the query string on the path will break things... or should it be the job of the cache to strip a potential query string if it wants to lookup on a filesystem?

Reply:
is the use case to allow requesting an image from a different server in a web request? or does this only make sense when coming through another path? if the later, we could make do the parse_url part in the controller instead of here...

Reply:
The use case is to be able to call the image http://myImgServer.com/image.png?v=123 from http://myAppServer.com/mypage
We need to maintain the whole path, including the parameter. Some images are served via specialized servers such as: https://images2.productserve.com/?w=200&h=200&bg=white&trim=5&t=letterbox&url=ssl%3Awww.cdiscount.com%2Fpdt2%2F9%2F2%2F5%2F1%2F700x700%2FNIN4902370537925.jpg&feedId=38753&k=421fa9c064b91b3719c999b1a900cfb30d7e23de

Reply:
yep, i got that. my question was more like "where is the image url coming from?". as one code path to get here is through the controller of this bundle that is asked for an image. i would not expect that we want that controller to get a domain passed and load random images from any 3rd party host...

so maybe we could move the parse_url logic into that controller instead, and say that if you use other means we expect you to specify exactly the file you need, but if you come through the controller, we make sure to reduce things to a path and strip domain or query string if they exist...

Reply:
Well, this was exactly my use case. I had loads of image paths (literally thousands) loaded into DB via a feed load. These images would then get displayed to the users but we wished to avoid relying on 3rd parties as much as possible so we built the cache along the way.
This was also a good way to deliver the image as we deemed fit with the change operations done when building the cache.
Before we made this fix we used to download the pictures when loading the data but it created other issues. Leveraging the cache and filters is really what made more sense.

Reply:
i spent some time with a collegue to try to understand what is going on.

unless we are completely mistaken, when the image is not in the local cache, it means that we provide the full image url in the path of the ImagineController::filterAction which then proceeds to download the image and cache it with the filter applied. i think that LiipImagineBundle should not support this - it would allow an attacker to provide the controller with an endless number of images from other domains. that way one could place all kinds of illegal content into the cache, or overload the cache.

did i misunderstand, or is this how the flow works?

i will look at documenting this better, but to me the change was a bug/security fix that had side effects for your use case. as it has been this way for a long time now, i think reverting it would not make sense.